### DISCUSSION TITLE
QR codes with undefined values?

### DISCUSSION THREAD

**Discussion by SgtSeppel** [gopro/labs#464] [Q&A]
I bought some laminated cards with QR codes to switch between settings during diving. 
In hindsight, I should probably have gotten myself familiar with the labs firmware to design and print my own codes. 

The problem I am having with the codes is, that on the most recent firmware (Hero11, March 23) the camera shuts down a few seconds after scanning a QR code. 
This seems to be related to the auto switch off setting. If I go to the menu quick enough, I can see that the value is empty. If I then move the value from the empty spot to i.e. 5 minutes, the camera does not shut down anymore - at least not after a few seconds. 
Downgrading to the December '22 firmware solves this particular issue, but another one remains.

I usually turn the signal tones off. Especially under water, they are very annoying. After scanning a QR code however, they are always turned back on. Since the camera is in a housing, I cannot access the touch screen to switch them back off. 

Can I somehow copy the command string from an existing QR code into the editor to manually adapt some of the values and create a new QR out of that?

**Comment by dnewman-gpsw** [comment]
QR Codes are just text, you smartphone can read them, and they are listed in the MISC/qrlog.txt file on your GoPro.  As this first issue sounds like bug, please share an image of the qr code that causes it.

**Comment by SgtSeppel** [comment]
I was assuming that this is a known issue. The company selling the cards posted a hint on their website about the issue. 
Here is a code that triggers the issue on the march23 firmware, but not on the december22 one:

![image](https://github.com/gopro/labs/assets/10022598/a0d8098c-c0dc-4992-befd-87c71724a20c)

I do not have the march23 firmware available right now, but if I go to the menu very quickly I can see that the value for the automatic shut down is empty and you then can scroll up (or down, not sure right now) to an actual value.

**Comment by SgtSeppel** [comment]
I'm trying to read more into the topics to dissect the string to find the offending command. 
After looking at the QR code generator I was assuming that maybe the camera is expecting a certain value, that is not set in the code and therefore it's leaving that undefined.

**Reply by dnewman-gpsw** [reply]
That is the case, the command string ends with oV7B4D2R1CS, this is now invalid. As the command language envolves, it needed to be more strict, as the CS part is ambiguous, it should be C0S0.  oCS command has a new meaning of HERO11 Mini, as full HERO11 doing a Mini command (short shutdown is the default on Mini.)
